src/mol-plugin-ui/structure/quick-styles.tsx

```diff
@@ -31,110 +33,131 @@ export class StructureQuickStylesControls extends CollapsableControls {
+
+            if (this.plugin.canvas3d) {
+                const p = PD.getDefaultValues(PostprocessingParams);
+                this.plugin.canvas3d.setProps({
+                    postprocessing: { outline: p.outline, occlusion: p.occlusion }
+                });
+            }
         }
     }
 
+    async changeValues(props: QuickStyleProps) {
+        const currentProps = this.state.props;
+        this.setState({ busy: true });
+        if (props.preset !== currentProps.preset) {
+            await this.setPreset(props.preset);
+        }
+        if (props.stylized !== currentProps.stylized) {
+            await this.setStylized(props.stylized);
+        }
+        this.setState({ props, busy: false });
+    }
+
+    async changePreset(preset: QuickStyleProps['preset']) {
+        await this.changeValues({ ...this.state.props, preset });
+    }
+
     render() {
-        return <div className='msp-flex-row'>
-            <Button noOverflow title='Applies default representation preset and sets outline and occlusion effects to default' onClick={() => this.default()} style={{ width: 'auto' }}>
-                Default
-            </Button>
-            <Button noOverflow title='Applies illustrative representation preset and Stylize it' onClick={() => this.illustrative()} style={{ width: 'auto' }}>
-                Illustrative
-            </Button>
-            <Button noOverflow title='Does not change representation, enables outline and occlusion effects, enables ignore-light representation parameter' onClick={() => this.stylized()} style={{ width: 'auto' }}>
-                Stylize Current
-            </Button>
-        </div>;
+        return <>
+            <div className='msp-flex-row'>
+                <ToggleButton label='Default' title='Applies default representation preset (depends on structure size)'
+                    toggle={() => this.changePreset('default')} isSelected={this.state.props.preset === 'default'} disabled={this.state.busy} />
+                <ToggleButton label='Cartoon' title='Applies cartoon polymer + ball-and-stick ligand representation preset'
+                    toggle={() => this.changePreset('cartoon')} isSelected={this.state.props.preset === 'cartoon'} disabled={this.state.busy} />
+                <ToggleButton label='Spacefill' title='Applies spacefill representation preset'
+                    toggle={() => this.changePreset('spacefill')} isSelected={this.state.props.preset === 'spacefill'} disabled={this.state.busy} />
+                <ToggleButton label='Surface' title='Applies molecular surface representation preset'
+                    toggle={() => this.changePreset('surface')} isSelected={this.state.props.preset === 'surface'} disabled={this.state.busy} />
+            </div>
+            <ParameterControls params={QuickStyleParams} values={this.state.props} onChangeValues={props => this.changeValues(props)} isDisabled={this.state.busy} />
```

Comment:
Here I wonder if this should be a toggle. Perhaps instead add two buttons 

![image](https://github.com/user-attachments/assets/7b7a6ba7-d8f7-42e0-869f-e0753b8ec12b)

This way you can actually remove the `componentDidMount` logic. And even thinking about that: when you click, it triggers the change, meaning the `isSelected={this.state.props.preset === '...'}` will never be applied.

So perhaps the UI can further be updated to

![image](https://github.com/user-attachments/assets/c9a65cb4-d9ad-4853-a9af-8e9ee3277821)

Reply:
After discussion with @dsehnal , I changed the Quick Styles UI to this:

<img width="296" alt="Screenshot 2025-01-17 at 11 21 09" src="https://github.com/user-attachments/assets/01eedd18-545d-4e9b-85b8-d99652433d8d" />

(It doesn't show what the current representation or style is, as that's tricky to keep track of.)